plugins/tileacq.py

```diff
@@ -568,6 +573,45 @@ def _check_fov(self, das, sfov):
             sfov = asfov
         return sfov
 
+    def _memory_check(self, _):
+        """
+        Makes an estimate for the amount of memory that will be consumed during
+        stitching and compares it to the available memory on the computer.
+        Displays a warning if memory exceeds available memory.
+        returns (bool): True (memory is sufficient), False (memory will be exhausted)
+        """
+
+        if self.stitch.value:
+            # Number of pixels for acquisition
+            pxs = 0
+            for s in self._get_acq_streams():
+                if hasattr(s, 'emtResolution'):
+                    pxs += s.emtResolution.value[0] * s.emtResolution.value[1]
+                elif hasattr(s, 'repetition'):
+                    pxs += s.repetition.value[0] * s.repetition.value[1]
+                else:
+                    logging.warning("Resolution of stream %s cannot be determined." % s)
+            pxs = pxs * self.nx.value * self.ny.value
+
+            # Memory calculation
+            mem_est = pxs * 2e-8  # Proportionality factor found empirically.
+            mem_computer = psutil.virtual_memory().total / 1024 ** 3  # in GB
+            # Assume computer is using 2 GB RAM for odemis and other programs
+            if mem_est < mem_computer - 2:
+                self._dlg.lbl_memory_check.SetLabel("")
+                self._dlg.Layout()
+                return True
+            else:
+                txt = "Stitching this area requires %.1f GB of memory.\n" % (mem_est) + \
+                    "Running the acquisition might cause your computer to crash."
+                self._dlg.lbl_memory_check.SetLabel(txt)
+                self._dlg.lbl_memory_check.SetFont(wx.Font(10, wx.DEFAULT, wx.NORMAL, wx.NORMAL))
+                self._dlg.lbl_memory_check.SetForegroundColour(wx.Colour(255, 80, 80))
+                self._dlg.Layout()
+                return False
+        else:
```

Comment:
Should reset the warning also when the stitching is disabled.
=> Change the code to first compute whether the memory is good or not, and, in a second part, update the label.